bundles/org.eclipse.ui/src/org/eclipse/ui/internal/SearchDecoration.java

```diff
@@ -14,17 +14,20 @@
 
 package org.eclipse.ui.internal;
 
+
 import java.util.regex.Pattern;
 import java.util.regex.PatternSyntaxException;
 
 import org.eclipse.jface.fieldassist.ControlDecoration;
 import org.eclipse.jface.fieldassist.FieldDecorationRegistry;
+import org.eclipse.swt.graphics.GC;
 import org.eclipse.swt.graphics.Image;
 
 /**
  * This class contains methods to validate and decorate search fields.
  */
 public class SearchDecoration {
+	private static GC gc;
```

Comment:
A static gc?

Reply:
Changed that, so that the GC is created within the method and then given as an Argument to getValidationError(). I hope that makes it better !

Reply:
Don't create it outside the method where it is actually used just to pass it as a parameter, it pollutes the code (see my comment below)